src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/DropColumns.cs

```diff
@@ -52,17 +63,31 @@ public override bool CheckTypes(CheckTypesContext context, TexlNode[] args, DTyp
             Contracts.AssertValue(errors);
             Contracts.Assert(MinArity <= args.Length && args.Length <= MaxArity);
 
-            var fArgsValid = base.CheckTypes(context, args, argTypes, errors, out returnType, out nodeToCoercedTypeMap);
-            Contracts.Assert(returnType.IsTable);
+            returnType = ReturnType;
+            nodeToCoercedTypeMap = null;
 
-            if (!argTypes[0].IsTable)
+            var isRecord = false;
```

Comment:
This code seems pretty shared, should Show/Drop columns be refactored to share a base class?

Reply:
Good idea, done